sql/updates/reverse-dev.sql

```diff
@@ -1,2 +1,49 @@
 -- Remove old compression policy procedures
 DROP PROCEDURE IF EXISTS _timescaledb_internal.policy_compression_execute(INTEGER, INTEGER, ANYELEMENT, INTEGER, BOOLEAN, BOOLEAN);
+
+DO $$
+DECLARE
+    caggs_nr int;
+BEGIN
+    SELECT count(*) FROM _timescaledb_catalog.continuous_agg WHERE bucket_width < 0 INTO caggs_nr;
+    IF caggs_nr > 0 THEN
+        RAISE EXCEPTION 'Downgrade is impossible since % continuous aggregates exist which use variable buckets', caggs_nr;
+    END IF;
+
+    -- It's save to drop the table.
+    -- ALTER EXTENSION is required to revert the effect of pg_extension_config_dump()
+    -- See "The list of tables configured to be dumped" test in test/sql/updates/post.catalog.sql
+    ALTER EXTENSION timescaledb DROP TABLE _timescaledb_catalog.continuous_aggs_bucket_function;
+
+    -- Actually drop the table.
+    -- ALTER EXTENSION only removes the table from the extension but doesn't drop it.
+    DROP TABLE _timescaledb_catalog.continuous_aggs_bucket_function;
+
+    -- Drop overloaded versions of invalidation_process_hypertable_log() and invalidation_process_hypertable_log()
+    -- with bucket_functions argument.
+
+    DROP FUNCTION _timescaledb_internal.invalidation_process_hypertable_log(
+        mat_hypertable_id INTEGER,
+        raw_hypertable_id INTEGER,
+        dimtype REGTYPE,
+        mat_hypertable_ids INTEGER[],
+        bucket_widths BIGINT[],
+        max_bucket_widths BIGINT[],
+        bucket_functions TEXT[]
+    );
+
+    DROP FUNCTION _timescaledb_internal.invalidation_process_cagg_log(
```

Comment:
I think those 2 should be `DROP FUNCTION IF EXISTS` for downgrades. At least that's what we have been doing.

Reply:
👌